File: src/gaussian_ffi.erl
-module(gaussian_ffi).
-export([set_document_class/1, get_preferred_color_scheme/0]).

set_document_class(_ClassName) ->
    % No-op for Erlang target (server-side)
    nil.

get_preferred_color_scheme() ->
    % Default to light for Erlang target (server-side)
    <<"light">>.

File: src/gaussian.ffi.mjs
export function setDocumentClass(className) {
  if (className === "dark") {
    document.documentElement.classList.add("dark");
    document.documentElement.classList.remove("light");
  } else {
    document.documentElement.classList.add("light");
    document.documentElement.classList.remove("dark");
  }
}

export function getPreferredColorScheme() {
  if (typeof window !== 'undefined' && window.matchMedia) {
    return window.matchMedia('(prefers-color-scheme: dark)').matches ? "dark" : "light";
  }
  return "light";
}
